ipaserver/plugins/cert.py

```diff
@@ -791,8 +793,10 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
                             "match requested principal") % gn.name)
             elif isinstance(gn, cryptography.x509.general_name.RFC822Name):
                 if principal_type == USER:
-                    if principal_obj and gn.value not in principal_obj.get(
-                            'mail', []):
+                    gn_value = (gn.value if isinstance(gn.value, list)
+                                         else [gn.value])
```

Comment:
In all tests that I did, `gn.value` was a string. However, since I don't know very much all about the code, I thought it would be a "safer" solution. Changed to your suggestion :)